code/modules/antagonists/cult/blood_magic.dm

```diff
@@ -564,90 +578,94 @@
-					candidate.undeploy()
-					candidate.emp_act(EMP_HEAVY)
-					var/construct_class = show_radial_menu(user, src, GLOB.construct_radial_images, custom_check = CALLBACK(src, PROC_REF(check_menu), user), require_near = TRUE, tooltips = TRUE)
-					if(!check_menu(user))
-						return
-					if(QDELETED(candidate))
-						channeling = FALSE
-						return
-					candidate.grab_ghost()
-					user.visible_message(span_danger("The dark cloud recedes from what was formerly [candidate], revealing a\n [construct_class]!"))
-					make_new_construct_from_class(construct_class, THEME_CULT, candidate, user, FALSE, T)
-					uses--
-					qdel(candidate)
-					channeling = FALSE
-				else
-					channeling = FALSE
-					candidate.color = prev_color
-					return
-			else
-				uses--
-				to_chat(user, span_warning("A dark cloud emanates from you hand and swirls around [candidate] - twisting it into a construct shell!"))
-				new /obj/structure/constructshell(T)
-				SEND_SOUND(user, sound('sound/effects/magic.ogg',0,1,25))
-				qdel(candidate)
-		else if(istype(target,/obj/machinery/door/airlock))
+/obj/item/melee/blood_magic/construction/cast_spell(atom/target, mob/living/carbon/user)
+	if(channeling)
+		to_chat(user, span_cult_italic("You are already invoking twisted construction!"))
+		return
+
+	var/turf/T = get_turf(target)
+	if(istype(target, /obj/item/stack/sheet/iron))
+		var/obj/item/stack/sheet/candidate = target
+		if(candidate.use(IRON_TO_CONSTRUCT_SHELL_CONVERSION))
+			uses--
+			to_chat(user, span_warning("A dark cloud emanates from your hand and swirls around the iron, twisting it into a construct shell!"))
+			new /obj/structure/constructshell(T)
+			SEND_SOUND(user, sound('sound/effects/magic.ogg',0,1,25))
+			return ..()
+		to_chat(user, span_warning("You need [IRON_TO_CONSTRUCT_SHELL_CONVERSION] iron to produce a construct shell!"))
+		return
+
+	if(istype(target, /obj/item/stack/sheet/plasteel))
+		var/obj/item/stack/sheet/plasteel/candidate = target
+		var/quantity = candidate.amount
+		if(candidate.use(quantity))
+			uses --
+			new /obj/item/stack/sheet/runed_metal(T,quantity)
+			to_chat(user, span_warning("A dark cloud emanates from you hand and swirls around the plasteel, transforming it into runed metal!"))
+			SEND_SOUND(user, sound('sound/effects/magic.ogg',0,1,25))
+			return ..()
+		return
+
+	if(istype(target,/mob/living/silicon/robot))
+		var/mob/living/silicon/robot/candidate = target
+		if(candidate.mmi || candidate.shell)
 			channeling = TRUE
-			playsound(T, 'sound/machines/airlockforced.ogg', 50, TRUE)
-			do_sparks(5, TRUE, target)
-			if(do_after(user, 5 SECONDS, target = user))
-				if(QDELETED(target))
+			user.visible_message(span_danger("A dark cloud emanates from [user]'s hand and swirls around [candidate]!"))
+			playsound(T, 'sound/machines/airlock_alien_prying.ogg', 80, TRUE)
+			var/prev_color = candidate.color
+			candidate.color = "black"
+			if(do_after(user, 9 SECONDS, target = candidate))
```

Comment:
if we're already re-arranging the code in this segment to look better, can you turn this do_after into a better early return? four indentations for everything that follows is 😵